### PR TITLE
feat: useCheckNftAccount in [LIB]

### PR DESCRIPTION
**Pull request by mcayuelas-ledger**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - ...

### 📝 Description

Hook added to `live-react-nft lib`

WIP for implementation in Gallery of account details on LLD

If Spam => add to HiddenCollections

https://github.com/user-attachments/assets/979a4572-a903-44aa-bbb6-fdf305232846

### ❓ Context

- **JIRA or GitHub link**: [LIVE-14031] 

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-14031]: https://ledgerhq.atlassian.net/browse/LIVE-14031?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-desktop/src/renderer/screens/nft/Collections/Collections.tsx [hunk 1]
```diff
@@ -71,11 +72,14 @@ const Collections = ({ account }: Props) => {
...
 
-  const { nfts, fetchNextPage, hasNextPage } = useNftGalleryFilter({
+  const { enabled, hideSpamCollection } = useHideSpamCollection();
+
+  const { nfts, fetchNextPage, hasNextPage } = useCheckNftAccount({
     nftsOwned: account.nfts || [],
     addresses: account.freshAddress,
     chains: [account.currency.id],
     threshold: isThresholdValid(thresold) ? Number(thresold) : 75,
+    action: enabled ? hideSpamCollection : undefined,
```
Comment (themooneer) [inline apps/ledger-live-desktop/src/renderer/screens/nft/Collections/Collections.tsx:82]: you can do it dynamically in one shot `...(enabled && {action: hideSpamCollection})`

**Review by themooneer — APPROVED** [review]

**Review by themooneer — APPROVED** [review]
👍

File: libs/live-nft-react/src/hooks/__tests__/useCheckNftAccount.test.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    const { cursor } = opts;
+    const index = cursor ? Number(cursor) : 0;
+
+    const pageIndex = Math.floor(index / pagedBy);
+    if (!apiResults[pageIndex]) throw new Error("no such page");
+
+    callCount++;
+    return Promise.resolve(apiResults[pageIndex]);
+  }),
+}));
```
Comment (KVNLS) [inline libs/live-nft-react/src/hooks/__tests__/useCheckNftAccount.test.tsx:57]: Can we use msw here?

Comment (LucasWerey) [reply libs/live-nft-react/src/hooks/__tests__/useCheckNftAccount.test.tsx:57]: We don't have msw in this lib
I can create a ticket to add it so we keep it in mind then update the test in the future if you want !

**Pull request merged**